app/sample_manifest_excel/sample_manifest_excel/upload/row.rb

```diff
@@ -19,8 +19,8 @@ class Row # rubocop:todo Metrics/ClassLength
       validates :number, presence: true, numericality: true
       validate :sanger_sample_id_exists?, if: :sanger_sample_id
       validates_presence_of :data, :columns
-      validate :check_sample_present
-      validate :sample_can_be_updated
+
+      validate :check_early_specialised_fields
```

Comment:
So there's a fair chunk of complexity coming out of the separation of early and late specialised fields. So just want to double check my understanding.

1) With the transaction moved outwards, this change is no longer actually required in order to fix the bug.
2) However it still results in the mini stock message being added to the transaction before the stock barcode is updated, which just feels a bit wrong
3) I think you also mentioned a worry that if we moved the generate stock event later and separate it from the register stock event, then we may end up with samples getting created, but not registered.

I'm not sure that the advantages of point 3 outweighs the complexity, especially when the counterpoint it we'd registering stocks that hadn't been created correctly, and so would be definition be faulty. (Such as what was happening before if validation failed.) 

Definitely a point of discussion though, but inclined to suggest moving `receptacle.register_stock!` to a callback after manifest processing.

Reply:
Yep I think that all makes sense. I think in addition to point 2, it's a feeling that future people looking at the code might not realise the importance of the transaction with delaying the registering of the stock resource, so it's better to make that explicit somehow. Agreed though, that it's actually a pain if the stocks get registered if the sample creation was faulty - so I'll have a quick look at moving that `register_stock` call to at the end.

Reply:
I've altered it substantially based on this - still waiting to see if tests pass, and doing some manual testing - but would appreciate another look at the last couple of commits to see if this matches what you were thinking. Thank you!